pkg/controller/direct/compute/forwardingrule_controller.go

```diff
@@ -0,0 +1,482 @@
+	}
+	if err != nil {
+		return fmt.Errorf("updating ComputeForwardingRule labels %s failed: %w", a.fullyQualifiedName(), err)
+	}
+	err = op.Wait(ctx)
+	if err != nil {
+		return fmt.Errorf("waiting ComputeForwardingRule %s update labels failed: %w", a.fullyQualifiedName(), err)
+	}
+	log.V(2).Info("successfully updated ComputeForwardingRule labels", "name", a.fullyQualifiedName())
+	// Get the updated resource
+	if a.Location == "global" {
+		getReq := &computepb.GetGlobalForwardingRuleRequest{
+			ForwardingRule: a.resourceID,
+			Project:        a.projectID,
+		}
+		updated, err = a.globalForwardingRulesClient.Get(ctx, getReq)
+	} else {
+		getReq := &computepb.GetForwardingRuleRequest{
+			ForwardingRule: a.resourceID,
+			Region:         a.Location,
+			Project:        a.projectID,
+		}
+		updated, err = a.forwardingRulesClient.Get(ctx, getReq)
+	}
+	if err != nil {
+		return fmt.Errorf("getting ComputeForwardingRule %q failed: %w", a.resourceID, err)
+	}
+
+	// setTarget request is sent when there are updates to target.
+	if !reflect.DeepEqual(forwardingRule.Target, a.actual.Target) {
+		if a.Location == "global" {
+			setTargetReq := &computepb.SetTargetGlobalForwardingRuleRequest{
+				ForwardingRule:          a.resourceID,
+				TargetReferenceResource: &computepb.TargetReference{Target: forwardingRule.Target},
+				Project:                 a.projectID,
+			}
+			op, err = a.globalForwardingRulesClient.SetTarget(ctx, setTargetReq)
+		} else {
+			setTargetReq := &computepb.SetTargetForwardingRuleRequest{
+				ForwardingRule:          a.resourceID,
+				TargetReferenceResource: &computepb.TargetReference{Target: forwardingRule.Target},
+				Project:                 a.projectID,
+				Region:                  a.Location,
+			}
+			op, err = a.forwardingRulesClient.SetTarget(ctx, setTargetReq)
+		}
+		if err != nil {
+			return fmt.Errorf("updating ComputeForwardingRule target %s failed: %w", a.fullyQualifiedName(), err)
+		}
+		err = op.Wait(ctx)
+		if err != nil {
+			return fmt.Errorf("waiting ComputeForwardingRule %s update target failed: %w", a.fullyQualifiedName(), err)
+		}
+		log.V(2).Info("successfully updated ComputeForwardingRule target", "name", a.fullyQualifiedName())
+	}
+
+	// Get the updated resource
+	if a.Location == "global" {
+		getReq := &computepb.GetGlobalForwardingRuleRequest{
+			ForwardingRule: a.resourceID,
+			Project:        a.projectID,
+		}
+		updated, err = a.globalForwardingRulesClient.Get(ctx, getReq)
+	} else {
+		getReq := &computepb.GetForwardingRuleRequest{
+			ForwardingRule: a.resourceID,
+			Region:         a.Location,
+			Project:        a.projectID,
+		}
+		updated, err = a.forwardingRulesClient.Get(ctx, getReq)
+	}
+	if err != nil {
+		return fmt.Errorf("getting ComputeForwardingRule %q failed: %w", a.resourceID, err)
+	}
+
+	status := &krm.ComputeForwardingRuleStatus{
+		LabelFingerprint:  updated.LabelFingerprint,
+		CreationTimestamp: op.Proto().InsertTime,
```

Comment:
do we want to update this ?

Reply:
This field is appearing in the realGCP log, so I think we should add it.

My bad I forgot to update this: `InsertTime` the insert of the operation, `CreationTimestamp` should be `updated.CreationTimestamp`